File: Recursion/binarySearch.py
def binary_search(arr, low, high, ele):
    if low <= high:
        mid = (low + high) // 2
        if ele == arr[mid]: return 'Found'
        elif ele < arr[mid]: return binary_search(arr, low, mid-1, ele)
        else: return binary_search(arr, mid+1, high, ele)
    else:
        return 'Not Found'

arr = [3,4,2,13,5]
arr.sort()
print(binary_search(arr, 0, len(arr) - 1, 5))

File: Recursion/binarySearch.js
const binarySearch = (arr, low, high) => {
  if (low <= high) {
    const mid = Math.floor((low + high) / 2);
    if (ele === arr[mid]) return mid;
    else if (ele < arr[mid]) return binarySearch(arr, low, mid - 1, ele);
    else return binarySearch(arr, mid + 1, high, ele);
  } else return -1;
};

const arr = [2, 4, 5, 9, 12, 15];
const ele = 12;
console.log(binarySearch(arr, 0, arr.length - 1, ele));
